<a href="https://colab.research.google.com/github/karye/Liu-labbar/blob/main/Lab3_AI_f%C3%B6r_L%C3%A4rare.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Laboration – AI-verktyg för lärare

Idag finns det många AI-verktyg som är fritt tillgängliga, förutsatt att du har ett Google-konto eller motsvarande. Dessa kan både du och dina elever enkelt få tillgång till. I denna laboration får du möjlighet att utforska några av dessa verktyg.

Laborationen är uppdelad i två delar:

1. **Del 1** – Du testar och bekantar dig med lättillgängliga AI-verktyg.
2. **Del 2** – Du tar mer kontroll och skapar material som känns relevant och intressant för dig.

Till varje uppgift följer ett antal reflektionsfrågor som du ska svara på under tiden du genomför den. I slutet av laborationen ska du även svara på avslutande frågor som rör laborationen i sin helhet.

> **OBS:** Det finns även andra, mer avancerade verktyg som inte är kostnadsfria. Ett av dessa kommer att presenteras under en demonstration den 30/3.

---


**Authors**

Joline Hellström, joline.hellstrom@liu.se

Cajsa Wargren, cajsa.wargren@liu.se


License

CC BY-NC-SA 4.0
https://creativecommons.org/licenses/by-nc-sa/4.0/

The lab (Notebook: code and instructions) is free to use, share, change and to make your own version suitable for your own students. The only restriction to this version and any new version is that 1) the same license (CC BY-NC-SA 4.0) is applied, 2) that the above mentioned authors are referred to as the authors of the original version and 3) that it cannot be used in a setting where a person or business has to pay to get access to, or in order to use, the version. I.e. full permission is given to use the lab, and subsequent versions, in schools or study circles, even running on paid-for computational resources (e.g. paid version of Google colab).

---

# Del 1

### Uppgift 1 – Skapa en egen hemsida

**Scenario:** Du ska hålla i AI-kursen för första gången. Eftersom kursen är ny sedan 2025 finns det inte många tillgängliga läromedel eller resurser. Därför vill du skapa en hemsida som eleverna kan använda för att lära sig mer på egen hand om AI-koncepten.

**Verktyg:** [ChatGPT](https://chatgpt.com/) och [Lovable](https://lovable.dev/dashboard)

#### Steg 1 – Skapa en prompt till Lovable med ChatGPT

För att generera en hemsida med Lovable behöver du en tydlig prompt. Du kan skriva den själv, men det är ofta effektivt att kombinera flera AI-verktyg. Be därför ChatGPT om hjälp genom att ge följande instruktion:

> *Det är dags för mig att hålla i kursen Artificiell Intelligens på gymnasiet (enligt GY25), nivå 1. Kan du ge förslag på olika delmoment som bör finnas med och formulera detta som en prompt som jag kan använda i Lovable för att skapa en hemsida eleverna kan arbeta med självständigt?*

#### Steg 2 – Skapa hemsidan med Lovable

1. Kopiera det svar du fick från ChatGPT.
2. Klistra in det som prompt i Lovable.
3. Lovable kommer nu att generera en hemsida åt dig. Detta kan ta någon minut.

#### Steg 3 – Skapa en egen designad prompt

För att anpassa hemsidan ytterligare efter dina önskemål (kring exempelvis färg, innehåll och struktur) kan du skriva en egen prompt till Lovable. Exempel:

> *Jag vill skapa en hemsida för att mina elever ska lära sig mer om AI. Den ska utgå från GY25 Artificiell Intelligens nivå 1 och innehålla både interaktiva uppgifter som quiz och kodceller som kan köras. Det ska finnas förklarande bilder och färgtemat ska följa Linköpings Universitets grafiska profil. Moduler med innehåll ska finnas på startsidan men det ska även finnas en ytterligare sida som visar ett schema för kursen och vilka uppgifter som eleverna ska genomföra i kursen samt när dessa har deadline.*


#### *Reflektionsfrågor*

**1.1** Hur såg din prompt ut i steg 3?  

**Svar:**

**1.2** Hur ser resultatet ut? Beskriv sidan och dess innehåll, och visa en skärmdump.  

**Svar:**

**1.3** Hur skiljer sig resultatet mellan att använda ChatGPTs prompt och din egna?  

**Svar:**


### Uppgift 2 – Generera en introduktionsvideo till AI-kursen

**Scenario:** Du har fått uppdraget att ansvara för den nya AI-kursen på ditt gymnasium. Eftersom kursen är ny och AI är ett aktuellt ämne, vill rektorn att du kort introducerar kursen för de andra tekniklärarna under ett möte. Mötet krockar dock med ditt schema, och du kan inte närvara fysiskt. Istället ska du skapa en film som introducerar kursen, som kan spelas upp i din frånvaro.

**Verktyg:** [NotebookLM](https://notebooklm.google.com/) och [HeyGen](https://app.heygen.com/home)

> **OBS:** Videogenerering är resurskrävande och kan ta några minuter att slutföra.

#### Steg 1 – Generera en video med NotebookLM

Ladda upp referensmaterial i NotebookLM, till exempel:

- Läroplanen GY25
- Ämnesplanen för AI-ämnet

Skriv sedan en så detaljerad prompt som möjligt om hur en introduktionsvideo till AI-kursen på gymnasiet kan se ut. Tänk på att:

- Utgå från centrala innehållet och de förmågor som specificeras i ämnesplanen.
- Videon ska ge en tydlig och pedagogisk introduktion till kursen.

#### Steg 2 – Generera en video med HeyGen

Skapa en prompt för videogenerering i HeyGen:

1. Du kan använda samma prompt som i steg 1 om du anser att den är tillräcklig.
2. Välj en avatar till videon.
3. Anpassa tema och stil för att ge videon önskad känsla.




#### *Reflektionsfrågor*

**2.1** Vilka prompts använde du?

**Svar:**

**2.2** Hur ser resultaten ut?

**Svar:**
    
**2.3** Vilka skillnader och likheter finns mellan dina videor?

**Svar:**
    
**2.4** Vilka för- och nackdelar ser du med de olika videogenereringsverktygen?

**Svar:**
    

---


### Uppgift 3 – Ljud till text: transkribera nyheter

**Scenario:** Du ska inleda lektionen med att arbeta med en nyhet där ni tillsammans lyssnar på ett nyhetsinslag. Du märker dock att vissa elever inte riktigt lyssnar aktivt eller snabbt glömmer innehållet. Istället för att spela upp klippet flera gånger och gå igenom alla frågor muntligt, vill du ha tillgång till innehållet som text.

**Verktyg:** Whisper (tal-till-text)

Idag finns det verktyg som kan omvandla tal till text och även översätta språk. Ett sådant verktyg är Whisper. Kod för att köra Whisper återfinns i kodcellerna nedan.

#### Steg 1 – Transkribera ett nyhetsinslag

Här laddar vi ner ett nyhetsinslag och skapar en textversion av det. Vill du använda ett annat inslag kan du byta ut länken i variabeln URL. Som standard används denna länk:

> <https://www.sverigesradio.se/artikel/krigsbloggare-rasar-nar-telegram-blockeras-i-ryssland>

Verktyget transkriberar ljudet till text och kan även översätta det.

> **OBS:** Transkriberingen kan ta lite tid.


In [ ]:
# --- Installs ---
!pip install -q yt-dlp openai-whisper fpdf2

# --- Imports ---
import whisper
import subprocess
import os
import textwrap
import requests
import re
from fpdf import FPDF
from pathlib import Path
from datetime import datetime

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 182.3/182.3 kB 1.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 803.2/803.2 kB 321.1 kB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 81.0/81.0 kB 829.1 kB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 366.9 kB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 327.1/327.1 kB 406.5 kB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━ 123.5/188.3 MB 411.5 kB/s eta 0:02:38

In [ ]:
URL = "https://www.sverigesradio.se/artikel/krigsbloggare-rasar-nar-telegram-blockeras-i-ryssland"
OUTPUT_DIR = "/content/sr_audio"
WHISPER_MODEL = "medium"  # "small" is faster, "medium"/"large" better for Swedish

os.makedirs(OUTPUT_DIR, exist_ok=True)

# ── 1. Scrape the audio URL from the article page ──────────────────────
page = requests.get(URL).text

audio_match = re.search(r'https?://www\.sverigesradio\.se/topsy/ljudfil/\d+', page)
if not audio_match:
    audio_match = re.search(r'"audioId\\?":\s*(\d+)', page)
    if not audio_match:
        raise RuntimeError("Could not find any audio on this article page.")
    audio_url = f"https://www.sverigesradio.se/topsy/ljudfil/{audio_match.group(1)}"
else:
    audio_url = audio_match.group(0).replace("\\", "")

print(f"Found audio URL: {audio_url}")

# ── 2. Download the MP3 (follow redirects to CDN) ──────────────────────
audio_path = os.path.join(OUTPUT_DIR, "audio.mp3")
resp = requests.get(audio_url, allow_redirects=True)
resp.raise_for_status()
with open(audio_path, "wb") as f:
    f.write(resp.content)
print(f"Audio downloaded: {audio_path} ({len(resp.content) / 1024:.0f} KB)")

# ── 3. Transcribe with Whisper ──────────────────────────────────────────
model = whisper.load_model(WHISPER_MODEL)
transcription = model.transcribe(audio_path, language="sv")
text = transcription["text"].strip()
print(f"\n--- Transcription ({len(text)} chars) ---\n{text}\n")

# ── 4. Build PDF ────────────────────────────────────────────────────────
FONT_PATH = "/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf"
if not os.path.exists(FONT_PATH):
    FONT_PATH = "/usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf"

pdf = FPDF()
pdf.add_page()
pdf.add_font("UniFont", "", FONT_PATH, uni=True)

pdf.set_font("UniFont", size=16)
pdf.cell(0, 12, "Sveriges Radio – Transcription", ln=True, align="C")

pdf.set_font("UniFont", size=9)
pdf.set_text_color(120, 120, 120)
pdf.cell(0, 8, f"Source: {URL}", ln=True, align="C")
pdf.cell(0, 6, f"Generated: {datetime.now().strftime('%Y-%m-%d %H:%M')}", ln=True, align="C")
pdf.ln(6)

pdf.set_text_color(0, 0, 0)
pdf.set_font("UniFont", size=11)

for paragraph in text.split("\n"):
    paragraph = paragraph.strip()
    if paragraph:
        pdf.multi_cell(0, 7, paragraph)
        pdf.ln(3)

pdf_path = os.path.join(OUTPUT_DIR, "transcription.pdf")
pdf.output(pdf_path)
print(f"PDF saved to: {pdf_path}")

# Auto-download in Colab
try:
    from google.colab import files
    files.download(pdf_path)
except ImportError:
    pass


#### Steg 2 – Testa med eget material

Välj ett eget ljud- eller videoklipp som är relevant för din undervisning och transkribera det med samma verktyg. Modifiera kodcellen ovan genom att ändra på den givna URL:en.

#### *Reflektionsfrågor*

**3.1** Hur väl stämmer transkriberingen överens med det som faktiskt sägs i inslaget?

**Svar:**

**3.2** Vilka för- och nackdelar finns det med att använda AI-översättningar?

**Svar:**

**3.3** Hur skulle du kunna använda detta verktyg i din roll som lärare?

**Svar:**

---


## Del 2

### Uppgift 4 – Skapa uppgifter med hjälp av AI

**Scenario:** Du vill effektivisera ditt arbete med att ta fram undervisningsmaterial och undrar hur AI kan hjälpa dig att skapa uppgifter som är kopplade till kursplanen.

**Verktyg:** [ChatGPT](https://chatgpt.com/) och [NotebookLM](https://notebooklm.google.com/)

#### Steg 1 – Formulera en detaljerad prompt

Välj ett moment eller delområde från en kurs du håller i. Skriv sedan en så detaljerad prompt som möjligt till ChatGPT där du beskriver:

- Vilken kurs och vilket centralt innehåll uppgiften ska koppla till
- Vilken nivå och typ av uppgift du vill ha (t.ex. resonemangsfråga, beräkningsuppgift, laborationshandledning)
- Vilka förkunskaper eleverna har
- Eventuella avgränsningar eller önskemål kring format

Testa att ge prompten till ChatGPT och granska resultatet. Ju mer specifik och genomtänkt din prompt är, desto mer användbart blir resultatet.

#### Steg 2 – Använd referensmaterial för att förbättra resultatet

Resultatet kan ofta förbättras om AI:n har tillgång till relevant referensmaterial. Ladda upp exempelvis tidigare prov, nationella prov (om sådana finns i ditt ämne), kursplanstexter eller andra dokument i NotebookLM. Be sedan verktyget att generera uppgifter baserat på det uppladdade materialet och jämför kvaliteten med det du fick i steg 1.

#### *Reflektionsfrågor*

**4.1** Hur ser den genererade uppgiften ut?

**Svar:**

**4.2** Resonera kring den genererade uppgiftens kvalitet.

**Svar:**

**4.3** Hur väl överensstämmer bedömningen med din egen värdering?

**Svar:**

---


### Uppgift 5 – Skapa egna bilder och visualiseringar

**Scenario:** Du vill skapa pedagogiska bilder och illustrationer som kan användas i din undervisning, utan att behöva rita eller designa dem själv.

**Verktyg:** [ChatGPT](https://chatgpt.com/) och [Microsoft Copilot](https://copilot.microsoft.com/)

#### Steg 1 – Testa två AI-verktyg

Välj två olika AI-verktyg för bildgenerering, till exempel ChatGPT och Microsoft Copilot. Använd denna prompt i båda verktygen:

> *Jag vill ha en bild över AI-utvecklingen som är pedagogisk för gymnasieelever.*

#### Steg 2 – Modifiera prompten

Omformulera prompten eller ge feedback på den genererade bilden. Exempel på saker du kan ändra:

- Färger
- Detaljnivå (mer eller mindre detaljer)
- Ljussättning
- Komposition (ändra vad som finns med eller tas bort)

Syftet är att förbättra bilden så att den blir tydligare och mer pedagogisk.

#### Steg 3 – Skapa en egen bild

Nu skapar du en helt egen illustration som passar för ett ämne eller moment du arbetar med (eller planerar att arbeta med). Tänk på:

- Färger och motiv
- Detaljnivå
- Pedagogisk tydlighet



#### *Reflektionsfrågor*

**5.1** Vilka AI-verktyg har du använt för att generera bilder?

**Svar:**

**5.2** Hur ser dina resultat ut? Klistra in de genererade bilderna från de olika stegen.

**Svar:**

**5.3** Ser du stora skillnader mellan bilder från olika verktyg? Vad beror det på?

**Svar:**

**5.4** Vilka regler gäller för att använda genererade bilder? Behöver man referera?

**Svar:**

**5.5** Hur ser du på användningen av genererade bilder i din roll som lärare? Hur ser du på att elever använder dem?

**Svar:**
    

---

### Uppgift 6 – Skapa en egen låt

**Scenario:** Du har försökt lära dina elever vad en derivata är, men märker att många ännu inte riktigt har förstått innehållet. Du upplever också att flera elever saknar intresse för matematik och att den traditionella undervisningen inte känns engagerande. Därför vill du testa ett kreativt arbetssätt: att introducera derivatan genom en låt, för att väcka elevernas intresse och göra innehållet mer minnesvärt.

**Verktyg:** [Suno](https://suno.com/)

#### Steg 1 – Generera en låt

Ge följande beskrivning till Suno:

> *Pedagogical hip-hop/rap song for high-school students:*
>
> *Vi har en kurva i ett koordinatsystem,*
> *den svänger upp och ner, men vad händer sen?*
> *Vill veta lutningen just här och nu,*
> *inte mellan två punkter — utan precis där du är, du.*
>
> *Vi krymper vårt steg, låter Δx gå mot noll,*
> *gränsvärdet visar vad förändringen har för roll,*
> *en tangentlinje som tangerar så fint,*
> *derivatan ger lutningen — exakt och precist.*
>
> *Derivata — förändringens takt,*
> *hur snabbt nånting ändras, det är vad vi har sagt.*

#### Steg 2 – Ändra genren

Modifiera prompten för att byta genre på låten, till exempel pop, rock, jazz eller elektronisk musik, beroende på dina egna preferenser.

#### Steg 3 – Anpassa låttexten

Ändra låttexten så att den handlar om ett annat område du arbetar med i undervisningen. Fundera på:

- Vilket ämne eller moment ska låten handla om?
- Hur kan du göra innehållet pedagogiskt och lättförståeligt?
- Hur kan texten bli engagerande och minnesvärd för eleverna?




#### *Reflektionsfrågor*

**6.1** Hur blev resultatet? Klistra in en länk till ljudfilen om möjligt.
    
**Svar:**

**6.2** Hur påverkas låtens pedagogiska effekt av genren och stilvalet?

**Svar:**

**6.3** Ser du några begränsningar eller utmaningar med att använda AI för detta ändamål?

**Svar:**

**6.4** Hur kan du använda AI-genererade låtar i din undervisning, och hur tror du att eleverna kan dra nytta av dem?

**Svar:**

---


## Uppgift 7 - Avslutande frågor

Svara på följande frågor som rör laborationen i sin helhet:

#### *Språkets påverkan*
**7.1** De flesta AI-verktyg presterar bäst på engelska. Testa att göra några av uppgifter fast med engelska istället för svenska (det vill säga använd engelska prompts och generera resultat på engelska). Märker du skillnad i kvalitet? Märkte du någon skillnad på resultatet?

**Svar:**

#### *AI-verktyg  i din roll som lärare*

**7.2** Hur skulle du kunna använda AI-verktyg i ditt dagliga arbete?

**Svar:**

**7.3** På vilket sätt kan AI-genererat material frigöra tid för mer kvalitativ lärartid med eleverna?

**Svar:**

**7.4** Kan AI hjälpa dig att våga testa undervisningsformer du annars inte hade hunnit utveckla?

**Svar:**

**7.5** På vilka sätt kan AI-verktyg förändra vad som förväntas av dig som lärare?

**Svar:**


**7.6** Vilka fördelar och begränsningar ser du med att använda AI för att skapa undervisningsmaterial?

**Svar:**

**7.7** Vilka fördelar och begränsningar ser du med att använda AI i bedömningsprocessen?

**Svar:**


**7.8** Hur kan AI-verktyg göra undervisningen mer tillgänglig för elever med olika behov?

**Svar:**

#### *AI-verktygens pålitlighet*

**7.9** Vilken typ av fel i AI-genererat material är svårast att upptäcka?

**Svar:**


**7.10** Vad är det största problemet du ser med att använda AI-genererat material rakt av utan granskning?

**Svar:**

#### *Användning i gymnasiekursen Artificiell Intelligens*

**7.11** Vilka av AI-verktygen från denna laboration anser du är aktuella för eleverna att arbeta med i AI-kursen? Motivera.

**Svar:**


**7.12** Hur stor vikt bör läggas på att eleverna ska använda aktuella AI-verktyg vs att eleverna själva ska implementera AI-lösningar på problem? Motivera.

**Svar:**


---


__*Bra jobbat, nu har du färdigställt labb 3!*__


